torch/csrc/jit/tensorexpr/llvm_codegen.cpp

```diff
@@ -513,34 +512,33 @@ void LLVMCodeGenImpl::emitKernel(
 
   irb_.CreateRet(value_);
 
-#if DEBUG_PRINT
-  llvm::errs() << *module_;
-#endif
   if (llvm::verifyFunction(*fn_, &llvm::outs())) {
     throw std::runtime_error("Function verification failed");
   }
 
-  // print graph debug info.
-  std::string fnstr;
-  llvm::raw_string_ostream FS(fnstr);
-  fn_->print(FS);
-  GRAPH_DEBUG("LLVM Function:\n", FS.str(), "\n");
+  // print graph debug info before optimization
+  llvm::SmallVector<char, 0> asmBuffer;
+  llvm::raw_svector_ostream asmStream(asmBuffer);
+  if (is_enabled(__FILE__, ::torch::jit::JitLoggingLevels::GRAPH_DEBUG)) {
+    module_->print(asmStream, nullptr);
+  }
+  GRAPH_DEBUG(asmStream.str().str(), "\n");
 
   optimize(*module_);
 
-#if DEBUG_PRINT
-  llvm::errs() << *module_;
-  llvm::SmallVector<char, 0> asmBuffer;
-  llvm::raw_svector_ostream asmStream(asmBuffer);
-  llvm::legacy::PassManager PM;
-  TM_->addPassesToEmitFile(
+  // print graph debug info after optimization
+  asmBuffer.set_size(0);
+  if (is_enabled(__FILE__, ::torch::jit::JitLoggingLevels::GRAPH_DEBUG)) {
```

Comment:
Nit: might be a good idea to wrap it with a macro and make it commonly available (if you decide to do that, please put that into a separate PR).

Reply:
Sounds good. Will go ahead and add it.

Reply:
To confirm, are you suggesting adding a macro in `jit/jit_log.h` and making it available to all code that includes this header file? Using this macro, we can disable some computation required for loggings when logging is not enabled.

Reply:
Yes, exactly that.

Reply:
Addressed in PR #49121.